spring-integration-file/src/main/java/org/springframework/integration/file/remote/AbstractRemoteFileStreamingMessageSource.java

```diff
@@ -180,27 +185,33 @@ public boolean isRunning() {
 	protected Object doReceive() {
 		Assert.state(this.running.get(), () -> getComponentName() + " is not running");
 		AbstractFileInfo<F> file = poll();
-		if (file != null) {
-			try {
-				String remotePath = remotePath(file);
-				Session<?> session = this.remoteFileTemplate.getSession();
+		while (file != null) {
```

Comment:
I think this is something wrong for this component contract.
We can't constantly poll from the source until `null`.
The point of the `receive()` is to get file or nothing.

Reply:
Below - if we are successful, the return ends the loop.